airflow/models/dagrun.py

```diff
@@ -621,10 +621,12 @@ def find_duplicate(cls, dag_id: str, run_id: str, *, session: Session = NEW_SESS
         return session.scalars(select(cls).where(cls.dag_id == dag_id, cls.run_id == run_id)).one_or_none()
 
     @staticmethod
-    def generate_run_id(run_type: DagRunType, logical_date: datetime) -> str:
+    def generate_run_id(
+        run_type: DagRunType, logical_date: datetime | None, run_after: datetime | None = None
```

Comment:
```suggestion
        *, run_type: DagRunType, logical_date: datetime | None, run_after: datetime | None = None
```

as the method becomes more complicate (hard to reason the order of logcial_date and run_after), we probably should make it keyword only

Reply:
I have changed it here: https://github.com/apache/airflow/pull/46616